blockchain/blockchain.go

```diff
@@ -323,6 +339,43 @@ func getBlockByHash(txn db.Transaction, hash *felt.Felt) (block *core.Block, err
 	})
 }
 
+// getBlockTxnCountByNumber retrieves the number of transactions in a block from database by its number
+func getBlockTxnCountByNumber(txn db.Transaction, number uint64) (count uint64, err error) {
+	numBytes := make([]byte, lenOfByteSlice)
+	binary.BigEndian.PutUint64(numBytes, number)
+
+	iterator, err := txn.NewIterator()
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		// Prioritise closing error over other errors
+		if closeErr := iterator.Close(); closeErr != nil {
+			err = closeErr
+		}
+	}()
+
+	var noOfTransactions uint64
+	prefix := db.TransactionsByBlockNumberAndIndex.Key(numBytes)
+	for iterator.Seek(prefix); iterator.Valid(); iterator.Next() {
```

Comment:
essentially duplicate of `getBlockByNumber`, we should eliminate the duplication